File: Android/app/src/main/java/com/example/smartfashion/model/Image.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Image(
    @SerializedName("image_id")
    val imageId: Int? = null,

    @SerializedName("user_id")
    val userId: Int,

    @SerializedName("url_original")
    val urlOriginal: String,

    @SerializedName("url_no_bg")
    val urlNoBg: String? = null,

    @SerializedName("storage_type")
    val storageType: String? = null,

    @SerializedName("created_at")
    val createdAt: Date? = null
)

File: Server/models/Image.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const imageSchema = new mongoose.Schema({
  image_id: {
    type: Number,
    unique: true 
  },
  user_id: {
    type: Number,
    required: true,
    ref: 'User',
    index: true 
  },
  url_original: {
    type: String,
    required: true
  },
  url_no_bg: {
    type: String,
    default: null 
  },
  storage_type: {
    type: String, 
    default: null 
  },
  created_at: {
    type: Date,
    default: Date.now
  }
});

imageSchema.pre('save', async function() {
  const doc = this;

  if (!doc.isNew) {
    return;
  }

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'image_id' }, 
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.image_id = counter.seq;
});

module.exports = mongoose.model('Image', imageSchema, 'images');